### PR TITLE
fix(billing): terminal/Machine substrate billing floors at 1.5x independent of AI markup

### PR DESCRIPTION
**Pull request by 2witstudios**
## Summary
- `calculateTerminalChargeCents` (`terminal-pricing.ts`) documented terminal/substrate billing's 1.5x floor but was dead code — its only caller was its own unit test. The real settle path charged 1.5x only because the shared `MARKUP_BPS` (also used for AI-model billing) happened to also default to 15000bps. If `MARKUP_BPS` were ever lowered or split per-source, terminal billing would silently drop below its documented floor with nothing to catch it.
- Added an independent `TERMINAL_MARKUP_BPS` constant (own env var, own default, clamped to a `TERMINAL_MARKUP_FLOOR_BPS = 15000` floor so a misconfigured env value can't silently violate it) and threaded it through as an explicit `markupBpsOverride` on `consumeCredits`/`AIMonitoring.trackUsage`, consumed by both terminal billing surfaces — active-runtime metering (`machine-billing.ts`) and idle-storage metering (`terminal-storage-billing.ts`), which had the identical gap.
- Deleted the dead function; every other billing caller (chat, voice, memory, workflows) is unaffected since the override defaults to the existing global `MARKUP_BPS` when absent.
- Added floor-enforcement tests proving the floor holds independent of `MARKUP_BPS`, not just formula consistency.

Tracked in PageSpace as node 4 (marked Done) on the new "Terminal — Tier Gating & Monetization" epic.

## Convergence history
- **Round 1 (Codex review):** flagged a crash-recovery gap — `credit-backfill.ts`'s orphan sweep (usage rows that reached `aiUsageLogs` but crashed before `consumeCredits` claimed a ledger row) recomputed the charge with no markup override, so terminal rows recovered via this path would silently bill at the shared `MARKUP_BPS` if it ever diverged from `TERMINAL_MARKUP_BPS`. Fixed by selecting `aiUsageLogs.source` in the sweep and reconstructing the override in `computeBackfillActions`. Also fixed a latent gap in the integration test's fake-DB harness that the new projection field exposed (its `aiUsageLogs` column mock was missing `source`, so its aggregate-query heuristic misfired).
- **Round 2 (self-review, 8-angle pass):** generalized `computeBackfillActions`'s markup parameter from a single hardcoded `source === 'terminal'` check to a `source -> bps` lookup map, so a second per-source floor is a one-line addition at the call site instead of a code change to the pure function. Typed `OrphanUsageRow.source` against the canonical `AIUsageSource` closed set instead of a bare string.
- **Round 3 (Codex re-review):** flagged that `TERMINAL_MARKUP_BPS` had no lower-bound validation — an env typo or someone mirroring a reduced `CREDIT_MARKUP_BPS` could silently drop terminal billing below the 1.5x floor the whole PR exists to enforce. Fixed by adding `TERMINAL_MARKUP_FLOOR_BPS = 15000` and clamping `TERMINAL_MARKUP_BPS = Math.max(envInt(...), floor)` — sub-floor values (including 0) are raised to the floor; deliberately higher values still pass through unclamped. New `credit-pricing.test.ts` covers the clamp, the zero-value edge case, unclamped pass-through, and independence from `CREDIT_MARKUP_BPS`.
- Merged `master` into this branch to resolve a `bun.lock` conflict (7 commits master had gained since this branch was cut — admin revamp, universal-commands chaining, OAuth/CORS/CSRF hardening, SDK 2.0.0). No other file conflicted.

## Test plan
- [x] `bunx vitest run` in `packages/lib` — 308/309 test files, 6870+ tests passing (full suite, not just touched files)
- [x] `bunx tsc --noEmit` in `packages/lib` and full monorepo `bun run typecheck` (Turbo, 16/16 tasks) — no new errors
- [x] `bun run lint` — 14/14 tasks clean
- [x] Branch merged with latest `master`, zero conflicts
- [x] CI green — Unit Tests, Lint & TypeScript Check, Security Test Suite, Dependency Audit, Secret Scanning, CodeQL, Static Security Analysis all passing

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01K4DfmudA4nGi6sBztYtBDS

### CODE REVIEW

File: packages/lib/src/monitoring/ai-monitoring.ts [hunk 1]
```diff
@@ -963,6 +969,7 @@ export async function trackAIUsage(data: AIUsageData): Promise<void> {
           // refreshes the right view; the navbar widget updates regardless.
           conversationId: data.conversationId,
           pageId: data.pageId,
+          markupBpsOverride: data.markupBpsOverride,
```
[the remark this answers is not in the document]
Comment (2witstudios) [reply packages/lib/src/monitoring/ai-monitoring.ts:972]: Fixed in 5b52cc144. `aiUsageLogs.source` already records `'terminal'` for both terminal billing surfaces, so the orphan sweep now selects `source` alongside the existing columns and `computeBackfillActions` stamps `markupBpsOverride: TERMINAL_MARKUP_BPS` onto the recovered action when `source === 'terminal'` — no schema migration needed. Added coverage in `credit-core.test.ts`, `credit-backfill.test.ts`, and an end-to-end case in `credits-flow.integration.test.ts` that drives the real `computeBackfillActions -> consumeCredits` path and asserts the recovered ledger row is stamped with `TERMINAL_MARKUP_BPS`, not the shared default.

File: packages/lib/src/billing/credit-pricing.ts [hunk 2]
```diff
@@ -140,6 +140,18 @@ export const TERMINAL_HOLD_ESTIMATE_CENTS = envInt('TERMINAL_HOLD_ESTIMATE_CENTS
...
+ * Markup applied to terminal/Machine substrate cost, in basis points — the
+ * founder-set floor of 1.5× real substrate cost. Deliberately a SEPARATE env
+ * var and constant from {@link MARKUP_BPS} (not derived from it): AI-model
+ * pricing and substrate-runtime pricing are different economics that may
+ * diverge, so lowering `CREDIT_MARKUP_BPS` for AI billing must never silently
+ * lower what terminal is charged. Threaded into `consumeCredits` via
+ * `markupBpsOverride` (see machine-billing.ts's `trackUsage`) so the real
+ * settle path enforces this floor independent of the shared default.
+ */
+export const TERMINAL_MARKUP_BPS = envInt('TERMINAL_MARKUP_BPS', 15000);
```
[the remark this answers is not in the document]
Comment (2witstudios) [reply packages/lib/src/billing/credit-pricing.ts]: Fixed in 6cdc0367f. Added `TERMINAL_MARKUP_FLOOR_BPS = 15000` and clamp `TERMINAL_MARKUP_BPS = Math.max(envInt('TERMINAL_MARKUP_BPS', floor), floor)` — a below-floor env value (typo, or someone mirroring a reduced `CREDIT_MARKUP_BPS`) is raised back to 15000, while a deliberately higher value still passes through unclamped (no stated ceiling). New `credit-pricing.test.ts` (module-reimport pattern, matching `auth/__tests__/constants.test.ts`) proves the clamp on a below-floor value, a zero value, pass-through of a higher value, and independence from `CREDIT_MARKUP_BPS` even when that's lowered.
